api/v1alpha1/linodemachine_types.go

```diff
@@ -75,6 +81,17 @@ type LinodeMachineSpec struct {
 	CredentialsRef *corev1.SecretReference `json:"credentialsRef,omitempty"`
 }
 
+// InstanceDisk InstanceDisks defines a list of disks to use for an instance
+type InstanceDisk struct {
+	// DiskID is the linode assigned ID of the disk
+	DiskID int `json:"diskID,omitempty"`
+	// SizeMB is the size in MB of the disk
+	// +kubebuilder:validation:Required
+	SizeMB int `json:"sizeMB"`
```

Comment:
Instead of embedding the value's units in the field name, maybe use a [resource.Quantity](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity) instead.

Reply:
the linode client requires size be in MB. I was thinking we explicitly call the unit out here(now in GB) so that we can simplify the conversions we will have to do to use the linode API

Reply:
Just for clarity (and to be regretfully pedantic), by "MB" do you mean _megabytes_ (decimal-based) or _mebibytes_ (binary-based)? I ask because the difference is kind of important. `10MB` (megabytes) is equivalent to `10 000 000 bytes`, where `10Mi` (mebibytes) is `10 486 760 bytes`. The binary SI units (mebibytes) are more accurate when it comes to describing disk space and memory sizes.

If the smallest a disk can be is 1Gi (gibibyte), that would be a separate concern; validating the value.

My suggestion for using resource.Quantity is mostly to reuse a type that is commonly used in Kubernetes to denote a quantity of some resource (in this case, disk). If the value needs to be coerced into another representation (e.g. binary SI &rarr; decimal SI) for the Linode API, resource.Quantity has methods to help with that.